### PR TITLE
Add a timeout argument to `progressWorkerEvent`

### PR DESCRIPTION
**Pull request by pentschev**
Instead of blocking indefinitely until an evens occurs, allows the user to specify a timeout the epoll wait operation, causing it to return.

In some cases using blocking mode in the progress thread may cause the event to never wake, allowing it to break periodically fixes hangs in those cases. It's not clear whether this is an abuse of blocking progress mode in UCX, but for now this resolves that issue.

### CODE REVIEW

**Review by wence- — COMMENTED** [review]
One style suggestions, otherwise LGTM.

File: cpp/src/worker.cpp [hunk 1]
```diff
@@ -244,18 +244,23 @@ void Worker::setProgressThreadStartCallback(std::function<void(void*)> callback,
...
-    pollingMode ? std::function<void()>{[]() {}} : std::bind(&Worker::signal, this);
+  std::function<bool()> progressFunction;
+  std::function<void()> signalWorkerFunction;
+  if (pollingMode) {
+    progressFunction     = [this]() { return this->progress(); };
+    signalWorkerFunction = []() {};
+  } else {
+    initBlockingProgressMode();
+    progressFunction = [this, epollTimeout]() { return this->progressWorkerEvent(epollTimeout); };
+    signalWorkerFunction = std::bind(&Worker::signal, this);
```
[the remark this answers is not in the document]
Comment (pentschev) [reply cpp/src/worker.cpp]: No objections, thanks for the suggestion.

**Review by wence- — APPROVED** [review]

**Comment by pentschev** [comment]
And btw, the Python benchmark timed out several times here, I now opened https://github.com/rapidsai/ucxx/pull/37 to increase the timeout for now.

**Pull request merged**